src/components/WeatherProvider.jsx

```diff
@@ -36,18 +36,17 @@ export default function WeatherProvider(props) {
     setCurrentWeather(current);
     setHourlyWeather(hourly);
     setDailyWeather(daily);
+    loading(false);
   };
 
   useEffect(() => {
-    if (geoData.latitude && geoData.longitude) {
```

Comment:
No need for conditional because useEffect occurs AFTER render, and so the coordinates are always present.